R/emfx.R

```diff
@@ -19,39 +19,51 @@ emfx = function(
     object,
     type = c("simple", "group", "calendar", "event"),
     post_only = TRUE,
+    intvar = NULL,
     ...
 ) {
   
+  # sanity check
+  if (!is.null(intvar)) {
+    if(!any(stringr::str_detect(rownames(object$coeftable), intvar))){
+      warning("The treatment was not interacted with \"intvar\" in the model object. Average margins are reported.")
+      intvar <- NULL
+      }
+  }
+  
   .Dtreat = NULL
   type = match.arg(type)
   gvar = attributes(object)[["etwfe"]][["gvar"]]
   tvar = attributes(object)[["etwfe"]][["tvar"]]
   gref = attributes(object)[["etwfe"]][["gref"]]
   tref = attributes(object)[["etwfe"]][["tref"]]
+  if(!is.null(intvar)) intvar = attributes(object)[["etwfe"]][["intvar"]]
   
   dat = eval(object$call$data, object$call_env)
   if (type=="event" & !post_only) {
     dat = dat[dat[[gvar]] != gref, , drop = FALSE]
   } else {
     if (".Dtreat" %in% names(dat)) dat = dat[dat[[".Dtreat"]], , drop = FALSE]
   }
-  
+
   if (type=="simple") by_var = ".Dtreat"
   if (type=="group") by_var = gvar
   if (type=="calendar") by_var = tvar
   if (type=="event") {
-    dat[["event"]] = dat[[tvar]] - dat[[gvar]]
+    dat[["event"]] = as.numeric(as.character(dat[[tvar]])) - as.numeric(as.character(dat[[gvar]]))
     by_var = "event"
   }
   
+  if(!is.null(intvar)) by_var <- c(by_var, intvar)
```

Comment:
Minor thing, but I'd prefer to use `=` for assignment instead of `<-` to be consistent.